src/tools/illink/test/Mono.Linker.Tests/TestCasesRunner/TrimmingTestLogger.cs

```diff
@@ -14,9 +16,9 @@ public TrimmingTestLogger ()
 			MessageContainers = new List<MessageContainer> ();
 		}
 
-		public List<MessageContainer> GetLoggedMessages ()
+		public ImmutableArray<MessageContainer> GetLoggedMessages ()
```

Comment:
To make sure we don't modify the message list outside of the logger, we should return an immutable array.